wsl-pro-service/internal/system/landscape.go

```diff
@@ -86,94 +121,60 @@ func (s *System) writeConfig(landscapeConfig string) (err error) {
-		return "", fmt.Errorf("could not parse config: %v", err)
+		return "", err
 	}
 
-	data.DeleteSection("host")
-
+	// Add or refresh computer title
 	distroName, err := s.WslDistroName(ctx)
 	if err != nil {
 		return "", err
 	}
-	if err := createKey(ctx, data, "client", "computer_title", distroName, true); err != nil {
-		return "", err
-	}
-
-	if err := createKey(ctx, data, "client", "hostagent_uid", hostagentUID, true); err != nil {
-		return "", err
-	}
-
-	if err := createKey(ctx, data, "client", "tags", "wsl", false); err != nil {
-		return "", err
+	oldComputerTitle, err := clientSection.GetKey("computer_title")
+	if err != nil {
+		if _, err = clientSection.NewKey("computer_title", distroName); err != nil {
+			return "", err
+		}
+	} else if oldComputerTitle.String() != distroName {
+		oldComputerTitle.SetValue(distroName)
 	}
 
-	if err := overrideSSLCertificate(ctx, s, data); err != nil {
+	// Refresh SSL certificate path if any
+	if err := overrideSSLCertificate(ctx, s, clientSection); err != nil {
 		return "", fmt.Errorf("could not override SSL certificate path: %v", err)
 	}
 
+	// Return the modified config as a string.
 	w := &bytes.Buffer{}
-	if _, err := data.WriteTo(w); err != nil {
-		return "", fmt.Errorf("could not write modified config: %v", err)
+	if _, err := iniFile.WriteTo(w); err != nil {
+		return "", fmt.Errorf("could not regenerate modified config: %v", err)
 	}
 
 	return w.String(), nil
 }
 
-// createKey tries to create a key with a particular value, optionally overriding an existing key.
-func createKey(ctx context.Context, data *ini.File, section, key, value string, override bool) error {
-	sec, err := data.GetSection(section)
-	if err != nil {
-		if sec, err = data.NewSection(section); err != nil {
-			return fmt.Errorf("could not find nor create section %q: %v", section, err)
-		}
-	}
-
-	if sec.HasKey(key) {
-		if !override {
-			log.Infof(ctx, "Landscape config contains key %q. Its value will not be overridden.", key)
-			return nil
-		}
-
-		log.Infof(ctx, "Landscape config contains key %q. Its value will be overridden.", key)
-		sec.DeleteKey(key)
-	}
-
-	if _, err := sec.NewKey(key, value); err != nil {
-		return fmt.Errorf("could not create key %q: %v", key, err)
-	}
-
-	return nil
-}
-
 // overrideComputerTitle converts the ssl_public_key field in the Landscape config
 // from a Windows path to a Linux path.
-func overrideSSLCertificate(ctx context.Context, s *System, data *ini.File) error {
-	const section = "client"
+func overrideSSLCertificate(ctx context.Context, s *System, section *ini.Section) error {
 	const key = "ssl_public_key"
 
-	sec, err := data.GetSection(section)
+	k, err := section.GetKey(key)
 	if err != nil {
-		// No certificate
+		// No certificate: nothing to transform
 		return nil
 	}
 
-	k, err := sec.GetKey(key)
-	if err != nil {
-		// No certificate
+	pathWindows := k.String()
+
+	if !strings.Contains(pathWindows, `\`) {
+		// Assuming this is a linux path already
```

Comment:
This is somewhat misleading. Windows paths can be written with forward-slashes, which complicates things. I think we should invoke wslpath unconditionally. An alternative would be to detect the regex `^\w:[\\\/].+` , assuming we only deal with absolute paths (Windows relative paths written with forward-slashes are Linux relative paths - any tool would break :) )

Here I miss Python's pathlib, where I can instantiate `PurePath` independent of the OS I'm working on.

Reply:
Also we should have a test case for that.

Reply:
We had a test case with a linux path and windows path (see coverage). But ok, it seems we’ll have to call wslpath with such a case for C://something. I’ll do it.